addons/l10n_id_efaktur/models/account_move.py

```diff
@@ -356,16 +355,51 @@ def _generate_efaktur(self, delimiter):
         my_utf8 = output_head.encode("utf-8")
         out = base64.b64encode(my_utf8)
 
-        attachment = self.env['ir.attachment'].create({
-            'datas': out,
-            'name': 'efaktur_%s.csv' % (fields.Datetime.to_string(fields.Datetime.now()).replace(" ", "_")),
-            'type': 'binary',
-        })
+        # can only generate efaktur doc IF:
+        # (1) no linked documents for all invoices; OR
+        # (2) all invoices are linked to same document AND document strictly links to these invoices as well;
+        attachment = self.env['ir.attachment']
+        if not self.l10n_id_efaktur_document:
+            self.l10n_id_efaktur_document = self.env['l10n_id_efaktur.document'].create({'invoice_ids': self.ids})
```

Comment:
I think you can add this field in the view after the number so that it shows there somewhere

Reply:
do you mean adding the l10n_id_efaktur_document or the attachment within it? I think including the efaktur-document record wouldn't be that useful?

Reply:
l10n_id_efaktur_document

I think it will become more useful later when we implement the EDI, and this document will hold information about it.
For now it doesn't hurt to have that there.